aws-iot-thinggroup/src/main/java/software/amazon/iot/thinggroup/Translator.java

```diff
@@ -53,67 +59,31 @@
  * mapping exceptions to appropriate Cloudformation exceptions
  */
 public class Translator {
-    static ProgressEvent<ResourceModel, CallbackContext> translateExceptionToProgressEvent(
-            ResourceModel model, Exception e, Logger logger) {
 
-        HandlerErrorCode errorCode = translateExceptionToErrorCode(e, logger);
-        ProgressEvent<ResourceModel, CallbackContext> progressEvent =
-                ProgressEvent.<ResourceModel, CallbackContext>builder()
-                        .resourceModel(model)
-                        .status(OperationStatus.FAILED)
-                        .errorCode(errorCode)
-                        .build();
-        if (errorCode != HandlerErrorCode.InternalFailure) {
-            progressEvent.setMessage(e.getMessage());
-        }
-        return progressEvent;
-    }
-
-
-    static HandlerErrorCode translateExceptionToErrorCode(Exception e, Logger logger) {
-        logger.log(String.format("Translating exception \"%s\", stack trace: %s",
-                e.getMessage(), ExceptionUtils.getStackTrace(e)));
-
-        // We're handling all the exceptions documented in API docs
-        // https://docs.aws.amazon.com/iot/latest/apireference/API_CreateThingGroup.html#API_CreateThingGroup_Errors
-        // (+same pages for other APIs)
-        // For Throttling and InternalFailure, we want CloudFormation to retry, and it will do so based on the error code.
-        // Reference with Retryable/Terminal in comments for each: https://tinyurl.com/y378qdno
+    static BaseHandlerException translateIotExceptionToHandlerException(
```

Comment:
just curious, do you know if these error codes  are outputted to CFN users directly or does CFN do additional translating?

Reply:
The translation we do here does model the IoT exceptions to corresponding CFN equivalent and will be thrown to the calling CFN service. That will be exposed to end-users via CFN internal logic of exposing these exception types.